src/deferredActions/GainProduction.ts

```
@@ -0,0 +1,34 @@
+import {Player} from '../Player';
+import {Resources} from '../Resources';
+import {DeferredAction, Priority} from './DeferredAction';
+import {LogBuilder} from '../LogBuilder';
+
+export namespace GainProduction {
+  export interface Options {
+    count?: number;
+    logMessage?: string;
+    logBuilder?: (builder: LogBuilder) => void;
+  }
+}
+
+export class GainProduction implements DeferredAction {
+  public priority = Priority.GAIN_RESOURCE_OR_PRODUCTION;
+  constructor(
+    public player: Player,
+    public resource: Resources,
+    public options: GainProduction.Options = {},
+  ) {}
+
+  public execute() {
+    if (this.options.count === undefined) {
+      this.options.count = 1;
+    } else {
+      this.options.count = Math.abs(this.options.count);
```

We can't force an argument's type to be something like a positive (or unsigned) integer.
So I either had to:
 - Throw an error if `count` was lower than 0
 - Make sure it's positive

I don't mind either way of doing it. Went with the latter since the deferred actions name are pretty clear, so negative number as argument would most likely be a simple mistake.

Let me know if you want this to change.

---

I would prefer to throw an error if `count` is not what we expect. This will fail faster and ensure a negative value doesn't indirectly end up being provided.

---

Done.